### ISSUE TITLE
pyright doesn't warn about possibly unbound `typing` types

### ISSUE THREAD

**Issue by Andrej730** [microsoft/pyright#8208]
See example, pyright doesn't warn about `Any` being possibly unbound though it does detect that it is unbound given the tooltip.

![image](https://github.com/microsoft/pyright/assets/9417531/c6eb9dfb-1560-44b6-bfe6-cebf29303966)

Not sure if this issue is very important - noticed that inconsistency accidentally when I was using `sys.version_info >= (3, x)` for the condition and it wasn't working because `sys` wasn't imported and then I've noticed that pyright didn't reported `Any` as unbound.

```python
# pyright: strict
a: bool = False

if a:
    from typing import Any
    from array import array
else:
    pass


# Expected "Any" is possibly unbound
# but there is no warning
def test() -> Any:
    pass


# "array" is possibly unbound
def test_2() -> array:
    pass
```

**Comment by erictraut** [comment]
Special forms like `Any` and `Literal` are tracked differently inside of pyright than classes. I agree this is a bug, but it would be very difficult and costly to address, and I don't think it's justified given that this is an edge case. I'm therefore going to close it as "won't fix".
